packages/abi-utils/lib/arbitrary.ts

```diff
@@ -0,0 +1,315 @@
+            fc.constant("nonpayable"),
+            fc.constant("payable")
+          ),
+          fc.boolean()
+        )
+        .map(([stateMutability, includeOptionals]) =>
+          !includeOptionals
+            ? { stateMutability }
+            : {
+                stateMutability,
+                payable: stateMutability === "payable",
+                constant:
+                  stateMutability === "view" || stateMutability === "pure"
+              }
+        )
+    )
+    .map(records => records.reduce((a, b) => ({ ...a, ...b }), {}))
+    .filter(entry => {
+      const { inputs, outputs = [] } = entry as Types.FunctionEntry;
+      // names that are not blank should be unique
+      const names = [...inputs, ...outputs]
+        .map(({ name }) => name)
+        .filter(name => name !== "");
+      return names.length === new Set(names).size;
+    });
+
+export const ReceiveEntry = () =>
+  fc.record({
+    type: fc.constant("receive"),
+    stateMutability: fc.constant("payable")
+  });
+
+export const FallbackEntry = () =>
+  fc
+    .tuple(
+      fc.record({
+        type: fc.constant("fallback")
+      }),
+      fc
+        .tuple(
+          fc.oneof(fc.constant("nonpayable"), fc.constant("payable")),
+          fc.boolean()
+        )
+        .map(([stateMutability, includeOptionals]) =>
+          !includeOptionals
+            ? { stateMutability }
+            : {
+                stateMutability,
+                payable: stateMutability === "payable",
+                constant:
+                  stateMutability === "view" || stateMutability === "pure"
+              }
+        )
+    )
+    .map(records => records.reduce((a, b) => ({ ...a, ...b }), {}));
+
+export const ConstructorEntry = () =>
+  fc
+    .tuple(
+      fc.record({
+        type: fc.constant("constructor"),
+        inputs: fc.array(Parameter())
+      }),
+      fc
+        .tuple(
+          fc.oneof(fc.constant("nonpayable"), fc.constant("payable")),
+          fc.boolean()
+        )
+        .map(([stateMutability, includeOptionals]) =>
+          !includeOptionals
+            ? { stateMutability }
+            : {
+                stateMutability,
+                payable: stateMutability === "payable",
+                constant:
+                  stateMutability === "view" || stateMutability === "pure"
+              }
+        )
+    )
+    .map(records => records.reduce((a, b) => ({ ...a, ...b }), {}));
+
+export const Abi = () =>
+  fc
+    .tuple(
+      ConstructorEntry(),
+      FallbackEntry(),
+      ReceiveEntry(),
+      fc.array(fc.oneof(FunctionEntry(), EventEntry()))
+    )
+    .chain(([constructor, fallback, receive, entries]) =>
+      fc.shuffledSubarray([constructor, fallback, receive, ...entries])
+    );
+
+namespace Numerics {
+  // 0 < n <= 32
+  export const Bytes = () => fc.nat(31).map(k => 32 - k);
+
+  // o < n <= 256
+  export const Bits = () => Bytes().map(k => 8 * k);
+
+  // 0 < n < 80
```

Comment:
This should be `0 < n <= 80` (matching the code you wrote below).

Well, OK -- I say 0<n<=80, but in fact the docs aren't fully consistent as to whether 0 is allowed or not, and when I asked the Solidity people about it, they said that's not fully resolved yet.  Anyway I think you're making the right call in excluding it for now.